src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from '../util/typed-inject';
+import { IRootStore } from 'app/stores';
+import { getFdaSubmissionNumber } from 'app/entities/companion-diagnostic-device/companion-diagnostic-device';
+import OncoKBTable from './OncoKBTable';
+import { IDeviceUsageIndication } from '../model/device-usage-indication.model';
+import { Column } from 'react-table';
+import { getCancerTypeName } from '../util/utils';
+import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';
+import CompactButton from '../button/CompactButton';
+import { SimpleConfirmModal } from '../modal/SimpleConfirmModal';
+
+interface CdxBiomarkerAssociationTableProps extends StoreProps {
+  editable?: boolean;
+  companionDiagnosticDeviceId: number;
+}
+
+export const CdxBiomarkerAssociationTable: React.FunctionComponent<CdxBiomarkerAssociationTableProps> = props => {
+  const [indications, setIndications] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [indicationToDeleteId, setIndicationToDeleteId] = useState(null);
```

Comment:
I don't think we need this state.

Reply:
It is to figure out which row we are deleting from the table. It is set when the delete button is clicked, so when we confirm in the the modal, we know which to delete.